apps/prairielearn/src/pages/instructorAssessmentInstance/instructorAssessmentInstance.html.ts

```diff
@@ -96,6 +97,50 @@ export function InstructorAssessmentInstance({
           navPage: '',
         })}
         <main id="content" class="container-fluid">
+          <div
+            class="modal fade"
+            id="resetQuestionVariantsModal"
+            tabindex="-1"
+            role="dialog"
+            aria-labelledby="resetQuestionVariantsModalLabel"
+          >
+            <div class="modal-dialog" role="document">
+              <div class="modal-content">
+                <div class="modal-header">
+                  <h4 class="modal-title" id="resetQuestionVariantsModalLabel">
+                    Confirm reset question variants
+                  </h4>
+                </div>
+                <div class="modal-body">
+                  <p>
+                    Are your sure you want to reset all current variants of this question for this
+                    ${resLocals.assessment.group_work ? 'group' : 'student'}?
+                    <strong>All ungraded attempts will be lost.</strong>
+                  </p>
+                  <p>
+                    This ${resLocals.assessment.group_work ? 'group' : 'student'} will receive a new
+                    variant the next time they view this question.
+                  </p>
+                </div>
+                <div class="modal-footer">
+                  <form name="reset-question-variants-form" method="POST">
+                    <input type="hidden" name="__action" value="reset_question_variants" />
+                    <input type="hidden" name="__csrf_token" value="${resLocals.__csrf_token}" />
+                    <input
+                      type="hidden"
+                      name="unsafe_instance_question_id"
+                      class="js-instance-question-id"
+                      value=""
```

Comment:
Tried copying over code from #8889 here. However, I am getting an error from this line here. Is there supposed to be a value here?

Reply:
That value should be getting filled in dynamically by this code: https://github.com/PrairieLearn/PrairieLearn/blob/7f0235a03f3f977deec29f69c03eb0c093a8fad5/apps/prairielearn/assets/scripts/instructorAssessmentInstanceClient.ts#L9 But I think this isn't working because above you are including `compiledScriptTag('instructorAssessmentInstance.ts')` which I think should actually be including `instructorAssessmentInstanceClient`? Hopefully there is a console error logged in the browser for this page at the moment?

The reason that value needs to be filled in by client-side JS code is that we only have a single modal for the whole page, but we trigger it from buttons on each question. We need the value in the modal form (the empty value you pointed out) to depend on which question button was clicked to trigger the modal. The client-side JS does this by reading the `data-instance-question-id` value from the button and setting this value into the modal form. This is a pattern that we use a lot with modals that can be triggered from different buttons.

Reply:
I see now. And yes, there was a console error for the missing page. I have updated to use `compiledScriptTag('instructorAssessmentInstanceClient.ts')` and the value now sets correctly.